packages/astro/src/core/build/internal.ts

```diff
@@ -37,9 +38,16 @@ export interface BuildInternals {
 
 	/**
 	 * A map for page-specific information.
+	 * // TODO: Remove in Astro 4.0
```

Comment:
Any reason to not use those directly now? It's internal, so doesn't really have to follow semver (unless it changes something for users downstream)

Reply:
I can't seem to find the type where integrations can access this. I've definitely changed this breakingly in the past because it should be internal for us 🤔 Do you know where it's exposed?

Reply:
https://github.com/withastro/astro/blob/3468c06f56447ee0aa4115b70f658780e6df75c5/packages/astro/src/core/build/static-build.ts#L237-L243

Reply:
Ah ok so it's `pagesByComponent` only. I'm also fine with deprecating it for now then, and maybe we can construct a better abstraction for Astro 4.0 later.